subsys/bluetooth/host/smp.c

```diff
@@ -2639,6 +2639,11 @@ void bt_set_bondable(bool enable)
 	bondable = enable;
 }
 
+bool bt_get_bondable(void)
+{
+	return bondable;
+}
```

Comment:
Does `ssp` use `smp`? 

If so, then `latch_bondable` should be used instead of this as our API supports setting `bondable` per connection.

Reply:
No. `ssp` does not use `smp`.

Reply:
Actually, `sm->bondable` is copy of `bondable`.
```
static bool latch_bondable(struct bt_smp *smp)
{
	(void)atomic_cas(&smp->bondable, BT_SMP_BONDABLE_UNINITIALIZED, (atomic_val_t)bondable);

	return atomic_get(&smp->bondable);
}
```

My change is also get a copy of `bondable`.

Reply:
> Actually, sm->bondable is copy of bondable.

`smp->bondable` can be overridden by `bt_conn_set_bondable`, so it's not always the same value as `bondable`

Reply:
> Then why are returning a `bondable` flag from SMP?

Arguably, the bondable setting does not belong in SMP. "Bondable mode" is a GAP thing.

So ideally, there should be a function named `bt_gap_set_bondable`.

A lower-level API that bypasses GAP, it should be named `bt_smp_set_bondable`. Not sure if it's a good idea to expose such a function though.

There should be no function with the non-specific name `bt_set_bondable` to minimize confusion.

Reply:
> Arguably, the bondable setting does not belong in SMP. "Bondable mode" is a GAP thing.

Then I guess the `bondable` value should also be placed outside of SMP (`smp.c`) (e.g. in `conn.c`)?

Reply:
Putting it in `conn.c` is OK, but I would prefer `gap.c`. That aligns better with how the core spec is organized.

Reply:
> > Actually, sm->bondable is copy of bondable.
> 
> `smp->bondable` can be overridden by `bt_conn_set_bondable`, so it's not always the same value as `bondable`

`ssp` has not a object likes as `smp`. For current implementation, function `bt_conn_set_bondable` is unsupported by BR/EDR. But the function `bt_conn_set_bondable` should be supported by BR/EDR. I will update PR to support it.

Reply:
> > Then why are returning a `bondable` flag from SMP?
> 
> Arguably, the bondable setting does not belong in SMP. "Bondable mode" is a GAP thing.
> 
> So ideally, the function would be named `bt_gap_set_bondable`. There should be no function with the non-specific name `bt_set_bondable` to minimize confusion.
> 
> Lower-level APIs that switch between SMP taking the value from GAP and a different source should be prefixed with `bt_smp_`.



> > Arguably, the bondable setting does not belong in SMP. "Bondable mode" is a GAP thing.
> 
> Then I guess the `bondable` value should also be placed outside of SMP (`smp.c`) (e.g. in `conn.c`)?



> Putting it in `conn.c` is OK, but I would prefer `gap.c`. That aligns better with how the core spec is organized.

I recommend making a separate PR for this change. I think we can create a task for it.

Reply:
> Putting it in `conn.c` is OK, but I would prefer `gap.c`. That aligns better with how the core spec is organized.

We don't have a `gap.c` yet, but I agree. A lot of what's inside of `conn.c` arguably belongs in `gap.c` (and ditto for `conn.h` and `gap.h`), but that'll be a larger change :)